src/_playground/documentation/Properties/_PropertyType.js

```diff
@@ -57,9 +57,14 @@ const PropertyType = ({ prop, componentName }) => {
                     <div>{`(${values.join('; ')})`}</div>
                 </React.Fragment>
             );
+        case 'shape':
         case 'i18n':
-            const shapeName = `${componentName} - Localized Text`;
-            return (<a href={`#${makeSafeId(shapeName)}`}>Localized Text</a>);
+            return (
+                <React.Fragment>
+                    <div>{prop.typeName}</div>
+                    {topLevel && <div>(<a href={`#${makeSafeId(getShapeTitle(componentName, propName, prop.typeName))}`}>details</a>)</div>}
```

Comment:
For now anyway, let's only show "details" link for shapes that are top-level prop types.  It could get messy in a hurry if shapes are embedded within out prop types.